tests/unit/test_project.py

```diff
@@ -68,10 +83,13 @@ def test_project_unmarshal(yaml_data):
     assert project.cmd == ["world"]
     assert project.env == [{"NAME": "VALUE"}]
     assert project.parts == {
-        "foo": {
-            "plugin": "nil",
-            "overlay-script": "ls",
-        }
+        **pebble_part,
```

Comment:
might be good to see a test where we override one value only too, what happens if a user declares "plugin: make"?

Reply:
I'm not sure I understand. can you elaborate pls? 

> what happens if a user declares "plugin: make"?

from which part?

Reply:
After some (offline) discussion with @sergiusens, the latest version of the code supports a pre-existing (user-provided) "pebble" part. If the part exists, we assume it's complete (so no trying to "merge" the two parts together). I think that's an OK compromise for now, and since we can revisit this in the future after pebble+rockcraft has matured a bit I'll resolve this. Thanks!